iset.mm

```diff
@@ -64371,6 +64378,85 @@ all reals (greatest real number) for which all positive reals are greater.
       GYQVEUUDYJYSFYGYQYIVHVFUUDYMUUAGYGYQYLVHVFVIYBYCYOEYDYE $.
   $}
 
+  $c |_|d $.
+
+  $( Syntax for the domain-disjoint-union of two relations. $)
+  cdjud $a class ( R |_|d S ) $.
+
+  $( The "domain-disjoint-union" of two relations: if ` R C_ ( A X. X ) ` and
+     ` S C_ ( B X. X ) ` are two binary relations, then ` ( R |_|d S ) ` is the
+     binary relation from ` ( A |_| B ) ` to ` X ` having the universal
+     property of disjoint unions.
+
+     Remark: the restrictions to ` dom R ` (resp. ` dom S ` ) are not necessary
+     since extra stuff would be thrown away in the post-composition with ` R `
+     (resp. ` S ` ), but they are explicitly written for clarity.  (Contributed
+     by MC and BJ, 10-Jul-2022.) $)
+  df-djud $a |- ( R |_|d S ) =
+            ( ( R o. `' ( inl |` dom R ) ) u. ( S o. `' ( inr |` dom S ) ) ) $.
+
+  ${
+    djufun.f $e |- ( ph -> Fun F ) $.
+    djufun.g $e |- ( ph -> Fun G ) $.
+    $( The "domain-disjoint-union" of two functions is a function.
+       (Contributed by BJ, 10-Jul-2022.) $)
+    djufun $p |- ( ph -> Fun ( F |_|d G ) ) $=
+      ( cdm cres ccnv ccom wfun cin c0 wceq wf1 wf df-f1 simprbi mp1i funco wss
+      cinl cinr cun cdjud cdju inlresf1 syl2anc inrresf1 crn dmcoss df-rn ss2in
+      sseqtr4i mp2an djuinN a1i syl5sseq ss0 syl syl21anc df-djud funeqi sylibr
+      funun ) ABUABFZGZHZIZCUBCFZGZHZIZUCZJZBCUDZJAVHJZVLJZVHFZVLFZKZLMZVNABJVG
+      JZVPDVEVEVIUEZVFNZWBAVEVIUFWDVEWCVFOWBVEWCVFPQRBVGSUGACJVKJZVQEVIWCVJNZWE
+      AVEVIUHWFVIWCVJOWEVIWCVJPQRCVKSUGAVTLTWAAVFUIZVJUIZKZVTLVRWGTVSWHTVTWITVR
+      VGFWGBVGUJVFUKUMVSVKFWHCVKUJVJUKUMVRWGVSWHULUNWILMAVEVIUOUPUQVTURUSVHVLVD
+      UTVOVMBCVAVBVC $.
+  $}
+
+  $( The domain of the "domain-disjoint-union" is the disjoint union of the
+     domains.  Remark: its range is the (standard) union of the ranges.
+     (Contributed by BJ, 10-Jul-2022.) $)
+  djudom $p |- dom ( F |_|d G ) = ( dom F |_| dom G ) $=
```

Comment:
Maybe this wants a name like `casedm` or something? (for one thing, `dm` rather than `dom` is the usual abbreviation for domain in theorem names).

I was thinking that `djudom` would naturally be the name for `( A ~<_ B /\ C ~<_ D ) -> ( A |_| C ) ~<_ ( B |_| D )` by analogy to http://us.metamath.org/mpeuni/undom.html